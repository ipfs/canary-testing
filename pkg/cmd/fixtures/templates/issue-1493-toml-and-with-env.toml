{{ define "simple-template" -}}

# this should receive the env
value="{{ $.Env.MyValue }}"
version="{{ .go_version }}"

{{- end -}}

[metadata]
name = "name"

{{- with (load_resource "./resource.toml") }}
  
# `withEnv` forward the `$.Env` value
{{ template "simple-template" (withEnv .) }}

# `toml` print the value as a toml.
# `pick` let's you pick a value from a resource.
{{ (pick . "values") | toml -}}

# atoi convert a string to a number
another_id="{{ (index .values (atoi "1")).id }}"

{{- end -}}