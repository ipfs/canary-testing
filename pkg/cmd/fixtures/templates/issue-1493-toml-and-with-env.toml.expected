[metadata]
name = "name"
  
# `withEnv` forward the `$.Env` value
# this should receive the env
value="123"
version="1.18"

# `toml` print the value as a toml.
# `pick` let's you pick a value from a resource.
[[values]]
  id = "a"

[[values]]
  id = "b"
# atoi convert a string to a number
another_id="b"